buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -217,25 +221,15 @@ class PerformanceTestPlugin : Plugin<Project> {
             delete(deferred { tasks.withType<JavaExecProjectGeneratorTask>().map { it.outputs } })
         }
 
-    private
-    fun Project.createPerformanceReportTask(performanceTestSourceSet: SourceSet): TaskProvider<PerformanceReport> =
-        tasks.register("performanceReport", PerformanceReport::class.java) {
-            systemProperties(propertiesForPerformanceDb())
```

Comment:
I think the properties for the performance DB should not be an input to the task. Especially the password should not be an input. Can we model that on the task instead of adding another bunch of system properties?

Reply:
I don't think we should remove it as system properties. What I said is that the password should not be an input.

Reply:
IIUC, you're concerning that a person might be able to get the password by comparing different builds on build scan. However, I still think these username and password shouldn't passed between system - why can't every build fetch it from TC environment? This will at least make our code less.

Reply:
Also changing the password should not make the build out-of-date/cause a cache miss. I don't see how using the environment variables makes the code any better. I think you want to fetch the password from the environment in the performance test code, right?

Reply:
Yes, I think every build can be able to fetch the password directly from environment so I don't see why we need to pass it between systems.

Reply:
If we add it as an environment variable then every process on the machine can access it. If we add the password as a system property, only the chosen java processes can access it.
That is why I prefer using system properties. Moreover, setting up a system property for a build is much easier for me (add something to my `gradle.properties`) than setting up an environment variable.